lib/jxl/enc_modular.cc

```diff
@@ -260,6 +260,9 @@ ModularFrameEncoder::ModularFrameEncoder(const FrameHeader& frame_header,
         prop_order.erase(prop_order.begin() + 1);
       }
     }
+    int max_properties = std::max<int>(
+        cparams_.options.max_properties,
+        frame_header.nonserialized_metadata->m.num_extra_channels + 2);
```

Comment:
I suppose a slightly better bound is possible in case the color channels are not modular encoded (then it would be -1 instead of +2)...